orttraining/orttraining/core/session/training_session.cc

```diff
@@ -86,6 +91,15 @@ Status SetupOptimizerParams(
     if (mixed_precision_weight_name_it != fp32_weight_names_to_mixed_precision_node_args.end()) {
       opt_node_config.mixed_precision_weight_arg = mixed_precision_weight_name_it->second;
     }
+
+    // check if initial optimizer states have been provided for weight
+    if (!init_optimizer_states.empty()){
+      const auto optim_state_it = init_optimizer_states.find(weight_name);
+      if (optim_state_it != init_optimizer_states.end()) {
+        opt_node_config.initial_states = std::move(optim_state_it->second);
```

Comment:
i got a bit confused by moving from a const iterator, but i guess the iterator itself is not a const_iterator.
would it be simpler to just copy from config.init_optimizer_states directly instead of making a copy into init_optimizer_states?